### PR TITLE
feature/design-system

### LINKED ISSUE #19: Bootstrap design system
Create SCSS entities for

• Colors
• Fonts

### PR DESCRIPTION
**Pull request by abdigali1**
Closes #19

if you do not know diff between SASS and SCSS, check this [link](https://stackoverflow.com/questions/5654447/whats-the-difference-between-scss-and-sass)(docs of SASS is not working right now)

Before pushing cheched hot relaod, global styling, test and git hooks. Everything is working)

- Add _typography.scss mixins and _fonts.scss with font declaration
-  Documentation `./docs/design-system.md`

- `./styles/__mocks__` is fix for jest that can't parse import .scss in .tsx files.  [ref is here](https://stackoverflow.com/questions/39641068/jest-trying-to-parse-css)

### CODE REVIEW

File: app/renderer/component/App.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var React = require('react');
+import { hot } from "react-hot-loader/root";
+import { Heading } from './components';
+
+export default hot((): JSX.Element =>
```
Comment (aidarnouman) [inline app/renderer/component/App.tsx]: hot -> App

File: static/styles/_colors.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// You can use it like color: $main_font;
+//main colors
+$red: #EC3944;
+$red_dark: #B6353D;
+$red_light: #F89299;
+//text colors
```
Comment (aidarnouman) [inline static/styles/_colors.scss]: • Add black whitelines before each beginning of the section 
• Add whitespace after `//`

File: static/index.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
+@import "./styles/colors";
+
+// @font-face {
```
Comment (aidarnouman) [inline static/index.scss]: Why didn't you add styles (mixins probably?) for fonts?
We have special typography style name for each of the fonts. You may look it up in figma (e.g. bold_20, bold_24 & etc.)

Comment (aidarnouman) [reply static/index.scss]: Also they need to be placed in separate file (`_typography.scss`)

File: tsconfig.json [hunk 4]
```diff
@@ -1,21 +1,30 @@
...
     "allowSyntheticDefaultImports": true,
     "strict": true,
     "forceConsistentCasingInFileNames": true,
-    "module": "es6",
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
+    "outDir": "./.webpack/main",
+    "module": "commonjs",
```
Comment (aidarnouman) [inline tsconfig.json]: You cannot change module resolution just like that - there should be solid reason to do that. Ask Nartay - he did some thorough research on module resolution system and may help you with that. But as far as I am concerned you shouldn't change it

File: webpack.rules.js [hunk 5]
```diff
@@ -1,32 +1,50 @@
-module.exports = [
+module.exports = rules = [
   // Add support for native node modules
   {
     test: /\.node$/,
-    use: 'node-loader',
-  },
-  {
-    test: /\.(js|jsx|ts|tsx)$/,
```
Comment (aidarnouman) [inline webpack.rules.js:8]: I think it was a proper setup. But you changed it to only `test: /\.tsx?$/,` on line 19

File: static/index.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+//     font-family: "NotoSansJP";
+//     src: url('static/fonts/NotoSansJP-Regular.ttf') format("truetype");
+// }
+// @mixin NotoSansJP {
```
Comment (aidarnouman) [inline static/index.scss]: Remove all of these unneeded comments here. Your code that goes to develop branch should never contain commented-out lines of code.

File: static/index.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+//     font-family: "NotoSansJP" ;
+// }
+
+body {
+  margin: auto;
+  max-width: 38rem;
+  padding: 2rem;
+  // @include NotoSansJP;
```
Comment (aidarnouman) [inline static/index.scss]: Change it with the mixin you are going to create from `_typography.scss`

File: app/renderer/component/App.tsx (hunk 1 above)
Comment (abdigali1) [reply app/renderer/component/App.tsx]: We can not change it, I think it is name of function or class

Comment (yerkeboolean) [reply app/renderer/component/App.tsx]: > We can not change it, I think it is name of function or class

`import { export1 as alias1 } from "module-name";`
[link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)

actually we can

File: package.json [hunk 8]
```diff
@@ -96,27 +99,31 @@
...
+    "prettier": "^2.0.5",
     "react-test-renderer": "^16.13.1",
+    "sass-loader": "^9.0.1",
     "spectron": "^10.0.1",
     "style-loader": "^0.23.1",
     "ts-jest": "^25.4.0",
-    "ts-loader": "^6.2.1",
+    "ts-loader": "^6.2.2",
     "ts-node": "^8.10.1",
-    "typescript": "^3.7.0",
```
Comment (murattishkul) [inline package.json:112]: were these changes introduced by eslint? if this is the case, I'd like to have them in different commit for better code review

**Comment by murattishkul** [comment]
Good job bro @abdigali1 !

**Review by nugmanoff — APPROVED** [review]

**Pull request merged**